arch/Kconfig

```diff
@@ -69,6 +69,16 @@ config USERSPACE
 	This feature is under heavy development and APIs related to it are
 	subject to change, even if declared non-private.
 
+config PRIVILEGED_STACK_SIZE
+	int "Size of privileged stack"
+	default 512
+	depends on ARCH_HAS_USERSPACE
+	help
+	This option sets the priviliged stack region size that will be cut
+	out of the user mode thread stack.  During normal execution, this
```

Comment:
Would it be possible to *add* this space to the thread stack, instead of cutting space out?
Otherwise, turning on user mode will have the effect of subtracting 512 bytes of stack space from every thread.
We want the app to work more or less the same if usermode is or is not enabled.

Reply:
It breaks power of 2 alignment, so no.  This would cause us to have to double the size of stacks and then waste a bunch of space on the end, which btw would be accessible by the thread.  The only way around that is to then define additional guards, which expends regions we can't afford to expend.

Reply:
We need to express this in Kconfig in some different way then. We need to make this really clear to users what is going on.

Not all MPUs enforce regions to be aligned to a multiple of their power-of-two size, and MMU-based systems just add an extra page. Is there some way we can set this up so that this 'carving out' behavior only takes place on systems with very strong region alignment constraints?

Maybe another Kconfig which controls if the kernel stack is added or carved out?

Reply:
this is more a function of the mpu features.  I think we need at least a Kconfig option for the alignment characteristic of the mpu.  Vanilla ARM mpus (ST and Beetle) need power of 2.  The NXP is the only one that only requires 32 byte alignment.  Mainly due to the fact they use a start/end pair of registers for defining the region.

Reply:
Adding another Kconfig for these characteristics is fine w/me.
Once this all lands, I can update the x86 port to use this value (and default to 4096) instead of hard-coding the kernel stack size, for sake of consistency.